app/components/Sidebar.tsx

```diff
@@ -55,16 +67,113 @@ const JumpToButton = () => {
 }
 
 export function Sidebar({ children }: { children: React.ReactNode }) {
+  const AnimatedDialogContent = animated(Dialog.Content)
+  const { isOpen } = useMenuState()
+  const config = { tension: 1200, mass: 0.125 }
+  const { pathname } = useLocation()
+
+  const transitions = useTransition(isOpen, {
+    from: { x: -50 },
+    enter: { x: 0 },
+    config: isOpen ? config : { duration: 0 },
+  })
+
   return (
-    <div className="flex flex-col border-r text-sans-md text-default border-secondary">
-      <div className="mx-3 mt-4">
-        <JumpToButton />
-      </div>
-      {children}
+    <>
+      {transitions(
+        ({ x }, item) =>
+          item && (
+            <Dialog.Root
```

Comment:
Feels slightly weird to be using a dialog here but it means we can use the same sidebar across both layouts. If there's a place that needs a bit more of a discerning eye it's probably here and the state logic to make sure it's bulletproof.